lib/rpc/globalRpcProviders.ts

```diff
@@ -0,0 +1,62 @@
+import { ChainId } from '@uniswap/sdk-core'
+import SingleJsonRpcProvider from './singleJsonRpcProvider'
+import UniJsonRpcProvider from './uniJsonRpcProvider'
+
+export default class GlobalRpcProviders {
+  private static readonly PROVIDER_RPC_URL_RANKING: Partial<Record<ChainId, number[] | undefined>> = {
+    [ChainId.MAINNET]: undefined,
+  }
+
+  private static readonly PROVIDER_RPC_URL_WEIGHTS: Partial<Record<ChainId, number[] | undefined>> = {
+    [ChainId.MAINNET]: undefined,
+  }
+
+  private static SINGLE_RPC_PROVIDERS: Partial<Record<ChainId, SingleJsonRpcProvider[]>> | null = null
+
+  private static UNI_RPC_PROVIDERS: Partial<Record<ChainId, UniJsonRpcProvider>> | null = null
+
+  private static initGlobalSingleRpcProviders() {
+    const INFURA_KEY = process.env.UNI_RPC_PROVIDER_INFURA_KEY
+    if (INFURA_KEY === undefined) {
+      throw new Error(`UNI_RPC_PROVIDER_INFURA_KEY must be a defined environment variable`)
+    }
+    const QUICKNODE_MAINNET_RPC_URL = process.env.UNI_RPC_PROVIDER_QUICKNODE_MAINNET_RPC_URL
+    if (QUICKNODE_MAINNET_RPC_URL === 'undefined') {
```

Comment:
I think you need to remove the quote for checking definedness. Otherwise you are checking `UNI_RPC_PROVIDER_QUICKNODE_MAINNET_RPC_URL` is not string literal `undefined`.